packages/application-generic/src/usecases/conditions-filter/conditions-filter.usecase.ts

```diff
@@ -37,31 +39,38 @@ import {
 import { EmailEventStatusEnum } from '@novu/stateless';
 import { PlatformException } from '../../utils/exceptions';
 import { createHash } from '../../utils/hmac';
-
+import { Instrument } from '../../instrumentation';
+import { CachedEntity } from '../../services/cache/interceptors/cached-entity.interceptor';
+import { buildSubscriberKey } from '../../services/cache/key-builders/entities';
 @Injectable()
 export class ConditionsFilter extends Filter {
   constructor(
+    private subscriberRepository: SubscriberRepository,
     private messageRepository: MessageRepository,
     private executionDetailsRepository: ExecutionDetailsRepository,
-    private subscriberRepository: SubscriberRepository,
     private jobRepository: JobRepository,
+    private tenantRepository: TenantRepository,
     private environmentRepository: EnvironmentRepository
   ) {
     super();
   }
 
-  public async filter(
-    command: ConditionsFilterCommand,
-    variables: IFilterVariables
-  ): Promise<{
+  public async filter(command: ConditionsFilterCommand): Promise<{
     passed: boolean;
     conditions: ICondition[];
+    variables: IFilterVariables;
   }> {
-    const { filters } = command;
+    const variables = await this.normalizeVariablesData(command);
+    const filters = command.filters?.length
+      ? command.filters
+      : command.step?.filters?.length
+      ? command.step.filters
+      : [];
```

Comment:
Not all of the clients have the job entity with filters (like select integrations), therefore we allow passing filters separately with priority.